benchmarks/engines/busyring/arbor/ring.cpp

```diff
@@ -145,42 +145,31 @@ class ring_recipe: public arb::recipe {
 
 struct cell_stats {
     using size_type = unsigned;
-    size_type ncells = 0;
-    size_type nsegs = 0;
+    size_type ncell = 0;
+    size_type nbranch = 0;
     size_type ncomp = 0;
 
     cell_stats(arb::recipe& r) {
-#ifdef ARB_MPI_ENABLED
-        int nranks, rank;
-        MPI_Comm_rank(MPI_COMM_WORLD, &rank);
-        MPI_Comm_size(MPI_COMM_WORLD, &nranks);
-        ncells = r.num_cells();
-        size_type cells_per_rank = ncells/nranks;
-        size_type b = rank*cells_per_rank;
-        size_type e = (rank==nranks-1)? ncells: (rank+1)*cells_per_rank;
-        size_type nsegs_tmp = 0;
-        size_type ncomp_tmp = 0;
-        for (size_type i=b; i<e; ++i) {
-            auto c = arb::util::any_cast<arb::cable_cell>(r.get_cell_description(i));
-            nsegs_tmp += c.num_segments();
-            ncomp_tmp += c.num_compartments();
-        }
-        MPI_Allreduce(&nsegs_tmp, &nsegs, 1, MPI_UNSIGNED, MPI_SUM, MPI_COMM_WORLD);
-        MPI_Allreduce(&ncomp_tmp, &ncomp, 1, MPI_UNSIGNED, MPI_SUM, MPI_COMM_WORLD);
-#else
-        ncells = r.num_cells();
-        for (size_type i=0; i<ncells; ++i) {
+        ncell = r.num_cells();
+
+        // Sample a subset of the cells to guess the number of branches
+        // and compartments in the entire model.
+        auto nsamp = std::min(size_type(10), ncell);
+        for (size_type i=0; i<nsamp; ++i) {
             auto c = arb::util::any_cast<arb::cable_cell>(r.get_cell_description(i));
-            nsegs += c.num_segments();
-            ncomp += c.num_compartments();
+            nbranch += c.morphology().num_branches();
+            // We discretise on samples, so the number of samples will approximate the
+            // number of samples
```

Comment:
Comment reads weirdly: should that be `number of compartments` at the end? If there were a way to distinguish more easily the samples in the morphology tree and the samples taken from the collection of cells, it would also aid clarity.

As part of my campaign to eradicate 'compartment' and replace with 'CV' or 'control volume', how do you feel about changing the terminology here, too?

Reply:
There was a type there, which I have fixed.

Agreed, and I changed the variable names. I didn't change the reporting, because the scripts that process benchmark output might depend on the exact wording.

Regarding name changes, I think that the argument for changing segment/section to branch is very clear because there was a lot of confusion in our old naming scheme, and a branch is well defined no.

Reply:
I certainly agree about the branch/segment nomenclature!